### PR TITLE
Add vector search e2e tests

### PR DESCRIPTION
**Pull request by gitbuda**
NOTE: To be able to run e2e tests, `cd tests && {{init_activate_ve3}} && ./setup.sh`
- [x] `./tests/e2e/vector_search` -> `cd build && make -j16 && ../tests/e2e/run.sh "Test behavior of vector search in Memgraph"` (activated ve3 and toolchain-v5)
- [x] `./tests/gql_behave/continuous_integration`
- [x] `tests/e2e/replication` -> `cd build && make -j16 && ../tests/e2e/run.sh "Replicate vector search"`
- [x] `tests/e2e/show_metrics` -> `cd build && make -j16 && ../tests/e2e/run.sh "Show metrics info"`
- [x] Add dummy mgbench -> `./tools/run_vector_search_index_to_remove.bash`
- [ ] Improve mgbench
    - [ ] Compare against no vector case (basic graph, simple node and edges)
    - [ ] Figure out why is throughput on `mgbench` low in general (saturate memgraph)
- [ ] Improve codebase + resolve TODOs
- [ ] Add more tests once all the existing ones pass
- [ ] Integrate all the tests (e.g. `mgbench` under CI)

### CODE REVIEW

File: tests/e2e/runner.py [hunk 1]
```diff
@@ -50,6 +50,7 @@ def load_workloads(root_directory):
     return workloads
 
 
+# TODO(gitbuda): Cleanup doesn't work properly if e2e test fails unexpectedly -> FIX.
```
Comment (gitbuda) [inline tests/e2e/runner.py:53]: Move this to another PR -> @hal-eisen-MG this might be (high probability) a source of flakyness.

**Comment by gitbuda** [comment]
### Tracking
- [ ] **[Link to Epic/Issue]**

### Standard development
- [ ] Update unit/E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch this branch

### CI Testing Labels
- [ ] Select the appropriate CI test labels _(CI -build=**build-name** -test=**test-suite**)_

### Documentation checklist
- [ ] Add the documentation label
- [ ] Add the bug / feature label
- [ ] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [ ] Write a release note, including added/changed clauses
    - **[ Release note text ]**
- [ ] **[ Documentation PR link memgraph/documentation#XXXX ]**
    - [ ] Is back linked to this development PR
- [ ] **[ Tag someone from docs team ]**

File: tests/mgbench/workloads/vector_search_index.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    def _get_random_vector(self):
+        return [random.random() for _ in range(0, VectorSearchIndex.VECTOR_DIMENSIONS)]
+
+    def benchmark__vector__single_vertex_lookup(self):
+        return ("MATCH (n:User {id:$id}) RETURN n;", {"id": self._get_random_node()})
+
+    def benchmark__vector__single_vertex_create(self):
+        i = self._nodes_count
+        self._nodes_count += 1
+        return ("CREATE (:Node {id:%i, vector: $vector});" % (i), {"vector": self._get_random_vector()})
```
Comment (DavIvek) [inline tests/mgbench/workloads/vector_search_index.py:63]: If you run the current benchmark, it may crash because the script we use to run the benchmark defines the size of the vector index based on the node count. During the benchmark, the node count increases as the benchmark task performs many CREATE operations, which add new entries to the index. This can cause the index size to be reached or even exceeded. To avoid this, we either need to hardcode the number of times the benchmark will run, or set a higher value for the index size limit.

Comment (gitbuda) [reply tests/mgbench/workloads/vector_search_index.py:63]: A quick step back here: try to think from the user's perspective how to handle the limit. Probably the v1 solution is to just expose the limit and document to the user what is happening. Once we have that 100% in place, here under benchmark, we just commit to something, e.g. putting the limit to a very high number (maybe that's something most users will do, so it might be representative to test for it).

**Comment by DavIvek** [comment]
Closing since this PR was merged into https://github.com/memgraph/memgraph/pull/2500
